airbyte-integrations/connectors/source-mongodb-v3/src/main/kotlin/io/airbyte/integrations/source/mongodb/MongoDbSourceConfiguration.kt

```diff
@@ -0,0 +1,150 @@
+                    MongoCredential.createCredential(
+                        clusterType.username,
+                        clusterType.authSource,
+                        clusterType.password.toCharArray(),
+                    )
+                schemaEnforced = clusterType.schemaEnforced
+            }
+            is SelfManagedReplicaSet -> {
+                connectionString = buildConnectionString(clusterType.connectionString)
+                if (
+                    clusterType.username.isNullOrBlank() ||
+                        clusterType.password.isNullOrBlank() ||
+                        clusterType.authSource.isNullOrBlank()
+                ) {
+                    mongoCredential = null
+                } else {
+                    mongoCredential =
+                        MongoCredential.createCredential(
+                            clusterType.username!!,
+                            clusterType.authSource!!,
+                            clusterType.password!!.toCharArray(),
+                        )
+                }
+                schemaEnforced = clusterType.schemaEnforced
+            }
+        }
+
+        val captureMode: CaptureMode =
+            when (pojo.updateCaptureMode) {
+                "Lookup" -> CaptureMode.LOOKUP
+                "Post Image" -> CaptureMode.POST_IMGAE
+                else -> throw ConfigErrorException("Invalid capture mode")
+            }
+
+        return MongoDbSourceConfiguration(
+            connectionString = connectionString,
+            maxConcurrency = 1, // todo: Add this configuration to the spec.
+            queueSize = pojo.queueSize ?: 10000,
+            discoverSampleSize = pojo.discoverSampleSize ?: 10000,
+            captureMode = captureMode,
+            resourceAcquisitionHeartbeat =
+                Duration.ofSeconds(pojo.initialWaitTimeInSeconds?.toLong() ?: 300),
+            mongoCredential = mongoCredential,
+            schemaEnforced = schemaEnforced,
+            initialLoadTime = Duration.ofHours(pojo.initialLoadTimeoutHours?.toLong() ?: 8),
+            initialWaitTime = Duration.ofSeconds(pojo.initialWaitTimeInSeconds?.toLong() ?: 300),
+            invalidCdcCursorPositionBehavior =
+                when (pojo.invalidCdcCursorPositionBehavior) {
+                    "Fail sync" -> InvalidCdcCursorPositionBehavior.FAIL_SYNC
+                    "Re-sync data" -> InvalidCdcCursorPositionBehavior.RESYNC_DATA
+                    else -> throw ConfigErrorException("Invalid invalidCdcCursorPositionBehavior")
+                },
+        )
+    }
+
+    enum class CaptureMode() {
+        LOOKUP,
+        POST_IMGAE
+    }
+
+    enum class InvalidCdcCursorPositionBehavior(val displayName: String) {
+        FAIL_SYNC("Fail sync"),
+        RESYNC_DATA("Re-sync data")
+    }
+    companion object {
+        const val DOUBLE_QUOTES_PATTERN = "\""
+        const val CREDENTIALS_PLACEHOLDER = "<username>:<password>@"
```

Comment:
I don't understand the purpose of this. Is this because of URL shenanigans where we don't want to expose the password in plain text in the UI?